### PR TITLE
Add dLocal gateway

### PR DESCRIPTION
**Pull request by curiousepic**
Unit:
15 tests, 45 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote:
21 tests, 59 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/d_local.rb [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+      def add_card(post, card, action, options={})
+        post[:card] = {}
+        post[:card][:holder_name] = card.name
+        post[:card][:expiration_month] = card.month
+        post[:card][:expiration_year] = card.year
+        post[:card][:number] = card.number
+        post[:card][:cvv] = card.verification_value
+        post[:card][:descriptor] = options[:dynamic_descriptor] if options[:dynamic_descriptor]
+        post[:card][:capture] = (action == 'purchase')
```
Comment (nfarve) [inline lib/active_merchant/billing/gateways/d_local.rb]: Should `post[:card][:capture]` always be passed?

File: test/fixtures.yml [hunk 2]
```diff
@@ -239,6 +239,12 @@ cyber_source:
   login: X
   password: Y
 
+# Working credentials, no need to replace
+d_local:
+  login: aeaf9bbfa1
+  trans_key: 9de3769b7e
+  secret_key: ae132899f56162a669b38dab5927862f3
```
Comment (nfarve) [inline test/fixtures.yml]: These are public credentials or Spreedly specific?

File: test/remote/gateways/remote_d_local_test.rb [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount-1, auth.authorization, @options)
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@amount, 'bad_id')
+    assert_failure response
+    puts response.inspect
```
Comment (nfarve) [inline test/remote/gateways/remote_d_local_test.rb]: Did you mean to keep this?

File: test/fixtures.yml (hunk 2 above)
Comment (curiousepic) [reply test/fixtures.yml]: We usually add them to AM, there's no reason to keep them private at least for dLocal. In any case, they're public now :)

Comment (nfarve) [reply test/fixtures.yml]: Okay. I thought the standard was to put a placeholder, unless they are values anyone uses when testing.

File: lib/active_merchant/billing/gateways/d_local.rb (hunk 1 above)
Comment (curiousepic) [reply lib/active_merchant/billing/gateways/d_local.rb]: I think it defaults to `false`, but there isn't much of a difference.

File: test/remote/gateways/remote_d_local_test.rb (hunk 3 above)
Comment (curiousepic) [reply test/remote/gateways/remote_d_local_test.rb]: Nice catch.

**Comment by nfarve** [comment]
Nice work! Really clean implementation.

**Comment by curiousepic** [comment]
Closed by 4ba8ec0789fd297b97a1c9cbfb118acabe534bbe

**Pull request closed without merging**